[FX] Speed up non-parameter tensor lookup

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#47325 [FX] Speed up non-parameter tensor lookup**

Differential Revision: [D24715484](https://our.internmc.facebook.com/intern/diff/D24715484)

Commit:
[FX] Speed up non-parameter tensor lookup

[ghstack-poisoned]

Commit:
Update on "[FX] Speed up non-parameter tensor lookup"


Differential Revision: [D24715484](https://our.internmc.facebook.com/intern/diff/D24715484)

[ghstack-poisoned]

dzhulgakov commented on torch/fx/symbolic_trace.py:
```diff
@@ -59,25 +59,19 @@ def create_arg(self, a: Any) -> Argument:
-            # Retrieve the qualname for an existing Tensor attribute
-            qualname_atoms : Optional[List[str]] = search_for_tensor(self.root)
-            qualname = '.'.join(qualname_atoms) if qualname_atoms else None
+            if not hasattr(self, 'tensor_attrs'):
```
hm, checking the attr is a bit hacky, maybe at least set it to None in the constructor?

In general - why not just move it to `trace` and run proactively? it seems like a better entry point

jamesr66a replied:
Yeah can move it to `trace()`

Commit:
Update on "[FX] Speed up non-parameter tensor lookup"


Differential Revision: [D24715484](https://our.internmc.facebook.com/intern/diff/D24715484)

[ghstack-poisoned]

zdevito commented on torch/fx/symbolic_trace.py:
```diff
@@ -59,25 +59,7 @@ def create_arg(self, a: Any) -> Argument:
-            # Retrieve the qualname for an existing Tensor attribute
-            qualname_atoms : Optional[List[str]] = search_for_tensor(self.root)
-            qualname = '.'.join(qualname_atoms) if qualname_atoms else None
+            qualname : Optional[str] = self.tensor_attrs[a] if a in self.tensor_attrs else None
```
nit: `self.tensor_attrs.get(a)`

Commit:
Update on "[FX] Speed up non-parameter tensor lookup"


Differential Revision: [D24715484](https://our.internmc.facebook.com/intern/diff/D24715484)

[ghstack-poisoned]
